modules/coredns/collect.go

```diff
@@ -38,11 +52,19 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 
 	mx := newMetrics()
 
-	cd.collectPanic(mx, raw)
-	cd.collectSummaryRequests(mx, raw)
-	//cd.collectSummaryRequestsDuration(mx, raw)
-	cd.collectSummaryRequestsPerType(mx, raw)
-	cd.collectSummaryResponsesPerRcode(mx, raw)
+	// some metric names are different depending on the version
+	// update them once
+	if !cd.skipVersionCheck {
+		cd.updateVersionDependentMetrics(raw)
+	}
```

Comment:
neat: let's change "cd.skipVersionCheck" to true right after "cd.updateVersionDependentMetrics(raw)", not inside it